bundles/org.eclipse.osgi/osgi/src/org/osgi/framework/ServiceException.java

```diff
@@ -80,6 +80,12 @@ public class ServiceException extends RuntimeException {
 	 * @since 1.8
 	 */
 	public static final int	ASYNC_ERROR			= 7;
+	/**
+	 * A deadlock occurred during service registration.
+	 * 
+	 * @since 1.8
+	 */
+	public static final int DEADLOCK = 8;
```

Comment:
This class is a copy of the OSGi Framework reference implementation and we don't want to modify it in Equinox-specific ways.
That being said I think it would be useful to have a dedicated error code for this situation so that SCR-Actors like org.apache.felix.scr could handle this situation gracefully.
But this has to be added to the OSGi spec first.